lib/classes/PluginManager.js

```diff
@@ -309,6 +310,27 @@ class PluginManager {
     });
   }
 
+  loadVariableGetters(pluginInstance) {
+    for (const [variablePrefix, variableGetter] of _.entries(
+      pluginInstance.variableGetters || {}
+    )) {
+      this.serverless.variables.customVariableResolverFuncs[variablePrefix] = variableGetter.bind(
+        this.serverless.variables
+      );
+      this.serverless.variables.variableResolvers.push([
+        new RegExp(`^${variablePrefix}:`),
+        variablePrefix,
+      ]);
+      if (variableGetter.dependentServiceName) {
```

Comment:
Here we introduce a contract to be used publicly, to which any changes in a future may mean a breaking change.

So acknowledging that, I would try to coin naming (for this configuration) with best what we can come up with at this stage, and ignore how we named things internally (that we can safely change later even with patch release).

--- 

I think `dependentServiceName` name is confusing - at first thought it's about stating on what service (to be named) this resolution depends.

I propose that:
1. We call this property a `serviceName` (as intention is to provide a human friendly name for this service, to be used in eventual user facing error message)
2. Fact that it's a _"dependent"_ (so that it depends on some core configuration, and resolution of variables relying on this services should be disabled at _prepopulation stage_), I would indicate with other property, which could be named  (some imperfect proposals): `isBaseConfigDependent`, `isCoreConfigDependent`, `isInitialConfigDependent`, `isDisabledAtPrepopulation` (anything better?)

It means we will need to define two properties in case of _"dependent"_ resolution, but I think readability of the outcome really pays back.

Reply:
Yeah, i was conflicted on what exactly to do here. combining your ideas, what about: 
```javascript
this.variableResolvers = {
  foo: resolveFoo,
  bar: {
    resolver: resolveBar
    serviceName: 'Some service',
    isDisabledAtPrepopulation: true
  }
}
```

Reply:
Looks great to me.

_(syntactically `resolver` may suggest object, that's why I personally would rather name a function `resolve`. Still that's a nit, and I'm aware that to many `resolver` sounds more natural,  I'm totally fine with it as well)_

Reply:
>  personally would rather name a function `resolve`

For an actual function(ie: simply defined and called directly), I would agree (see `resolveBar`) but for an assignment to an variable, passing it around, it makes more sense to me as `resolver`. I'm not sure why.

Reply:
I'm totally open to changing it. @pmuens @alexdebrie any thoughts?

Reply:
`resolver` is fine to me